src/pydata_sphinx_theme/__init__.py

```diff
@@ -83,7 +84,9 @@ def update_config(app, env):
         )
 
     # check the validity of the theme swithcer file
-    if isinstance(theme_options.get("switcher"), dict):
+    is_dict = isinstance(theme_options.get("switcher"), dict)
+    is_tested = not theme_options.get("check_switcher") is False
+    if is_dict and is_tested:
```

Comment:
```suggestion
    should_test = theme_options.get("check_switcher", False)
    if is_dict and should_test:
```

This suggestion seems a little cleaner to me... though I'm curious why you did this as an `html_theme_options` key and didn't take @coretl's suggestion to use an environment variable instead?  That seems more flexible to me, as you can leave `conf.py` the same and still get different behavior on different systems (e.g., local build, CI server, deploy server...)

Reply:
I'm happy with a theme option too, as I could always put in conf.py:
```python
# Don't check switcher if running on CI so we can bootstrap the creation
check_switcher = not os.environ.get("CI", False)
```

Reply:
Ok if @coretl is satisfied then no need to use env var.